### DISCUSSION TITLE
Add Fluent Ribbon QuickAccess item in code behind

### DISCUSSION THREAD

**Discussion by Rickyd56** [fluentribbon/Fluent.Ribbon#1211] [Q&A] [answered]
I am requesting help on adding a quickaccess item dynamically.  I have reviewed the code examples but, didn't see what I am trying to do. Your guidance is greatly appreciated.  Thanks Rick

I have a Fluent ribbon with quick access items.  I am looking to add an item to the bar in my Mainwindow.cs code.  Below is code that I have tried and it add a "Test" item with icon to the menu, not the quick access bar.  And, I am struggling on what the command would be (it is commented out).

Code in MainWindow.CS:

            QuickAccessMenuItem AddMenuItem = new QuickAccessMenuItem();
            AddMenuItem.Visibility = Visibility.Visible;
            AddMenuItem.Icon = "images/button.png";
            AddMenuItem.IsChecked = true;
            AddMenuItem.Header = "Test";
            Ribbon1.QuickAccessItems.Add(AddMenuItem);
//            AddMenuItem.CommandBindings = new CommandBindingCollection();

Quickaccess xml in the MainWindow.xml:

<Fluent:Ribbon.QuickAccessItems>
    
    <Fluent:QuickAccessMenuItem IsChecked="true">
        <Fluent:Button x:Name="btnBrowser1" Header="Browser" Icon="Images/Browser Button.bmp" Click="btnBrowser_Click" />
    </Fluent:QuickAccessMenuItem>
 
    <Fluent:QuickAccessMenuItem IsChecked="true">
        <Fluent:Button x:Name="btnExplorer1" Header="Explorer" Icon="Images/IntelliFile.png"  Click="btnExplorer_Click" />
    </Fluent:QuickAccessMenuItem>
    <Fluent:QuickAccessMenuItem IsChecked="true">
        <Fluent:Button x:Name="btnPaint1" Header="Paint" Icon="Images/MS Paint icon.png" Click="btnPaint_Click" />
    </Fluent:QuickAccessMenuItem>
    <Fluent:QuickAccessMenuItem IsChecked="False">
        <Fluent:Button x:Name="btnPaint2a" Header="Paint2" Icon="Images/MS Paint icon.png" Click="btnPaint2_Click" />
    </Fluent:QuickAccessMenuItem>
    <Fluent:QuickAccessMenuItem IsChecked="False">
        <Fluent:Button x:Name="btnPaint3a" Header="Paint3" Icon="Images/MS Paint icon.png" Click="btnPaint3_Click" />
    </Fluent:QuickAccessMenuItem>

    <Fluent:QuickAccessMenuItem IsChecked="true">
        <Fluent:Button x:Name="btnNotes1" Header="Notes" Icon="Images/note.png" Click="btnNotePad_Click" />
    </Fluent:QuickAccessMenuItem>
    <Fluent:QuickAccessMenuItem IsChecked="true" Visibility="Hidden">
        <Fluent:Button x:Name="qat_ProgTool1" Header="Zip Tool" Click="qatProgTool1_Click" />
    </Fluent:QuickAccessMenuItem>

    <Fluent:QuickAccessMenuItem IsChecked="true" Visibility="Hidden">
        <Fluent:Button x:Name="qat_ProgTool2" Header="Zip Tool" Click="qatProgTool2_Click" />
    </Fluent:QuickAccessMenuItem>

        
    <Fluent:QuickAccessMenuItem  />
</Fluent:Ribbon.QuickAccessItems>

**Comment by batzen** [comment]
To get an item to show up in the toolbar, not only the menu, you have to set a target on the menu item.
Your command etc. should then be set on that target.

**Comment by Rickyd56** [comment]
Batzen,

I appreciate your support.  Do you have an example of setting up a quickaccess menu item in code behind? And, I was wondering as well how do you save the quickaccess setup after a user changes it?  

I am new to Fluent Ribbon and again appreciate the help.   I have reviewed your showcase and didn't see this example.  

Rick

**Comment by Rickyd56** [comment]
Batzen,

I was able to connect the click event to the dynamic menu item, but still need help making it show up on the toolbar.  Can you please show me in code behind how to set the target?  Below is what I currently have without the target being set in MainWindow.cs.

                QuickAccessMenuItem AddMenuItem = new QuickAccessMenuItem();
                Fluent.Button AddButton = new Fluent.Button();

                AddMenuItem.Click += new RoutedEventHandler(btnProgram1_Click);
                AddMenuItem.Visibility = Visibility.Visible;
               // AddMenuItem.Target -> "{Binding ElementName = MenuConnect}";
                
                AddMenuItem.Icon = "images/button.png";
                AddMenuItem.IsChecked = true;
                AddMenuItem.Header = "Test";

                Ribbon1.QuickAccessItems.Add(AddMenuItem);

**Reply by batzen** [reply, marked as answer]
There is no need to set the icon etc. on the menu item directly.
Just use
```
QuickAccessMenuItem AddMenuItem = new QuickAccessMenuItem();
Fluent.Button AddButton = new Fluent.Button();
AddButton .Header = "Test";
AddButton .Icon = "images/button.png";
AddButton .Click += new RoutedEventHandler(btnProgram1_Click);

AddMenuItem.Target = AddButton;

AddMenuItem.IsChecked = true;

Ribbon1.QuickAccessItems.Add(AddMenuItem);
```

That should work.
Instead of adding a click handler you could also use a command.

**Comment by Rickyd56** [comment]
Batzen

Thank you very much.  It worked great and so simple.  

Rick
